File: fluentcore/src/main/java/com/mcreater/fxfluent/controls/icon/FluentIcon.java
package com.mcreater.fxfluent.controls.icon;

import javafx.scene.control.Label;
import javafx.scene.text.Font;

public class FluentIcon extends Label {
    private static final Font FLUENT_ICON_FONT = Font.loadFont(FluentIcon.class.getClassLoader().getResourceAsStream("fonts/Segoe Fluent Icons.ttf"), 10);
    private static final Font FLUENT_ICON_FONT2 = Font.loadFont(FluentIcon.class.getClassLoader().getResourceAsStream("fonts/Segoe Fluent Icons.ttf"), 14);
    public FluentIcon(char c) {
        this(c, false);
    }
    public FluentIcon(char c, boolean isB) {
        super(String.valueOf(c));
        setFont(isB ? FLUENT_ICON_FONT2 : FLUENT_ICON_FONT);
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/controls/icon/FluentIcon.kt
package com.mcreater.fxfluent.controls.icon

import javafx.scene.control.Label
import javafx.scene.text.Font

class FluentIcon(chr: Char?, size: Double?): Label() {
    init {
        font = Font.loadFont(
            FluentIcon::class.java.getClassLoader().getResourceAsStream("fonts/Segoe Fluent Icons.ttf"),
            size!!
        )
        text = String(charArrayOf(chr!!))
    }
    constructor(chr: Char?) : this(chr, 10.0)
}